downloadMarcxmlRecords.py

```diff
@@ -0,0 +1,59 @@
+import requests
+from bs4 import BeautifulSoup
+import csv
+import secrets
+import time
+import datetime
+import argparse
+import os
+
+# command line arguments
+parser = argparse.ArgumentParser()
+parser.add_argument('-f', '--fileName', help='the file of data to be searched. \
+optional - if not provided, the script will ask for input')
+args = parser.parse_args()
+
+if args.fileName:
+    fileName = args.fileName
+else:
+    fileName = input('Enter the file of data to be searched: ')
+
+# run time start, establish variables,
+startTime = time.time()
+
+wskey = secrets.wskey
+baseURL = 'http://www.worldcat.org/webservices/catalog/content/'
+
+with open(fileName) as csvfile:
+    reader = csv.DictReader(csvfile)
+    rowCount = len(list(reader))
+
+# script content
+dwnldDir = 'oclcRecords/'
+if os.path.isdir(dwnldDir) is False:
+    os.makedirs(dwnldDir)
+with open(fileName) as csvfile:
+    reader = csv.DictReader(csvfile)
+    for row in reader:
+        if rowCount != 0 and rowCount % 200 == 0:
```

Comment:
I assume this pausing is done to comply with some OCLC rate limit policy? Might be useful to mention/link to that in a comment or the README.

Reply:
It's more just the product of some tweaking to avoid the random timeouts errors I was encountering with larger requests but I will make a note in the readme